Patches/CheckGameEndPatch.cs

```diff
@@ -244,13 +238,13 @@ public static bool Prefix()
                     }
                     if (!Options.PhantomSnatchesWin.GetBool())
                     {
-                        //Phantom
-                        if (pc.Is(CustomRoles.Phantom) && !pc.IsAlive() && pc.GetPlayerTaskState().IsTaskFinished)
-                        {
-                            CustomWinnerHolder.WinnerIds.Add(pc.PlayerId);
-                            CustomWinnerHolder.AdditionalWinnerTeams.Add(AdditionalWinners.Phantom);
-                        }
+                    //Phantom
+                    if (pc.Is(CustomRoles.Phantom) && !pc.IsAlive() && pc.GetPlayerTaskState().IsTaskFinished)
+                    {
+                        CustomWinnerHolder.WinnerIds.Add(pc.PlayerId);
+                        CustomWinnerHolder.AdditionalWinnerTeams.Add(AdditionalWinners.Phantom);
                     }
+                }
                     //自爆卡车来咯
                     if (pc.Is(CustomRoles.Provocateur) && Main.Provoked.TryGetValue(pc.PlayerId, out var tar))
                     {
```

Comment:
This part of the code is not clear to me, did you close the loop earlier than necessary?
because Provocateur is also in the array

Reply:
I didn't close the loop. I just deleted one frame.